src/Application/Actions/RegularGivingMandate/Get.php

```diff
@@ -36,19 +36,24 @@ protected function action(Request $request, Response $response, array $args): Re
         if (! $this->environment->isFeatureEnabledRegularGiving()) {
             throw new HttpNotFoundException($request);
         }
+
+        Assertion::keyExists($args, "mandateId");
+        $mandateId = $args["mandateId"];
+        Assertion::string($mandateId);
         if (empty($args['mandateId'])) {
-            throw new DomainRecordNotFoundException('Missing donation ID');
+            throw new DomainRecordNotFoundException('Missing mandate ID ' . $mandateId);
         }
 
+        // @todo-regular-giving: make sure that the donor requesting the mandate is the owner of the mandate
         $donorId = $request->getAttribute(PersonWithPasswordAuthMiddleware::PERSON_ID_ATTRIBUTE_NAME);
         \assert($donorId instanceof PersonId);
         $uuid = Uuid::fromString((string) $args['mandateId']);
         $mandate = $this->regularGivingMandateRepository->findOneByUuid($uuid);
-
-        if (!$mandate) {
-            throw new DomainRecordNotFoundException('Mandate not found');
+        \assert($mandate !== null);
+        \assert($donorId === $mandate->donorId);
+        if ($donorId !== $mandate->donorId) {
+            throw new DomainRecordNotFoundException('Donor Id: ' . strval($donorId) . ' on request does not match donor Id on the mandate: ' . strval($mandate->donorId));
```

Comment:
strval is a sort of aggressive way to make sure you've got a string. It's better to work with the object by doing `$donorId->id`